mesonbuild/modules/rust.py

```diff
@@ -184,7 +185,7 @@ def test(self, state: ModuleState, args: T.Tuple[str, BuildTarget], kwargs: Func
             new_target_kwargs
         )
 
-        test = self.interpreter.make_test(
+        test: Test = self.interpreter.make_test(
```

Comment:
You shouldn't need the explicit type set, do you? Mypy should know that its `Test`

Reply:
It knows that it's a TestClass, but doesn't know that it's Test. Same for Doctest in the patch that introduces it.

Reply:
It's generic, it knows what type `klass` is, and it know that the return type is `instanceof<klass>`:

```python
from __future__ import annotations
from dataclasses import dataclass
import typing as T


@dataclass
class Foo:

    arg: str


@dataclass
class Bar(Foo):

    arg2: int = 0


_T = T.TypeVar('_T', bound=Foo)


def func(val: str, type_: T.Type[_T] = Foo) -> _T:
    return type_(val)


a = func('foo')
T.reveal_type(a)
b = func('bar', Bar)
T.reveal_type(b)
```
which if you run it tells you:
```
Runtime type is 'Foo'
Runtime type is 'Bar'
```

Reply:
I will double check but I am almost sure that it failed the CI job for mypy.

Reply:
The case with defaults does not work for me ("error: Need type annotation for "test") on either mypy 1.8 or 1.14.1. The type annotation for Doctest is there for the similarity and because overall it doesn't hurt.